code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -158,6 +159,10 @@
 	update_icon()
 	add_fingerprint(user)
 
+/obj/item/melee/baton/throw_impact(mob/living/carbon/human/hit_mob)
+	if(!..() && turned_on && istype(hit_mob))
+		thrown_baton_stun(hit_mob)
+	return ..()
```

Comment:
this will run the parent proc twice, do we want this?
might be better to do a `. = ..()` and do checks + return on `.`

Reply:
Should be changed, finally using the github app